infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/Action.java

```diff
@@ -0,0 +1,25 @@
+package org.corfudb.infrastructure.orchestrator;
+
+import org.corfudb.runtime.CorfuRuntime;
+
+/**
+ *
+ * An idempotent workflow action. All workflow actions must extend this class.
+ *
+ * Created by Maithem on 10/25/17.
+ */
+
+public abstract class Action {
+
+    /**
+     * Returns the name of this action.
+     * @return Name of action
+     */
+    public abstract String getName();
+
+    /**
+     * Execute the action.
+     * @param runtime A that the action will use to execute
+     */
+    public abstract void execute(CorfuRuntime runtime);
```

Comment:
Should these actions be able to pass state around somehow?

Reply:
My incoming patches will add a registry to persist and pass around a workflow's state.
For now, its stateless and actions are completely independent.

Reply:
Well, you're defining the interfaces, so the decision needs to be made now.
Each action should probably have a way to export state to the next action without going through the registry.

Reply:
I'll return a Map<String, Object> to model the state that needs to be exported.